python/unblob/handlers/executable/elf.py

```diff
@@ -31,6 +36,48 @@
 
 @attrs.define(repr=False)
 class ElfChunk(ValidChunk):
+    def upx_get_loader(self, file: File) -> int:
+        file.seek(-4, io.SEEK_END)  # last 4 bytes indicates where linfo ends
+        return convert_int32(file.read(4), Endian.LITTLE) - 12
+
+    def upx_calc_checksum(self, file: File, l_info: int, elf) -> bool:
+        file.seek(l_info, io.SEEK_SET)
+        l_checksum = convert_int32(file.read(4), Endian.LITTLE)
+        file.seek(l_info + 8, io.SEEK_SET)
+        l_lsize = convert_int16(file.read(2), Endian.LITTLE)
+        size_pack2 = elf.last_offset_segment - l_lsize
+        size_aligment = size_pack2  # Calc size before alignment
+        while size_aligment % 4 != 0:  # Forces to be mod 4 because of alignment
+            size_aligment += 1
+        xct_off = False  # For shared android library -> .so
+        for section in elf.sections:
+            if section.name == ".init":
+                xct_off = True
+                break
+        size_aligment += (
+            0 if xct_off else (4 & size_aligment) ^ (int(bool(xct_off)) << 2)
+        )  # 0 or 4
+        size_aligment += 8  # Added 2 times 4 byte (size of disp)
+        if xct_off:
+            size_aligment += 12
+        alignment = size_aligment - size_pack2
+        checksum_offset = elf.last_offset_segment - (l_lsize - alignment)
+        file.seek(checksum_offset, io.SEEK_SET)
+        return zlib.adler32(bytearray(file.read(l_lsize - alignment)), 1) == l_checksum
```

Comment:
You don't need to convert to `bytearray`, the function accepts bytes. Also if `l_size` is large, I would iterate over the file in chunks. `adler32` supports rolling checksums by setting the initial value (first time it's 1 but then you feed it the output of your first calculation).